### PR TITLE
Revising desktop release process

### PR DESCRIPTION
**Pull request by amyblais**
Before submitting, please confirm you've
 - [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [ ] executed `npm run lint:js` for proper code formatting

Please provide the following information:

**Summary**
Updating desktop app release checklist and adding details about our updated desktop app release cycle/schedule expectations.
Removing changelog from the desktop app repo.

### CODE REVIEW

**Comment by esethna** [comment]
Thanks @amyblais, Looks good as most items are similar to webapp process. 

Dev should review too as the technical aspects are different, adding @deanwhillier

File: docs/release-process.md [hunk 1]
```diff
@@ -12,70 +18,83 @@ Notes:
...
+### C. (T-minus 15 working days) Judgment Day
 
-### C. (T-minus 7 working days) Code Complete and Release Candidate Cut
+Day when Release Manager and PMs decide which major features are included in the release, and which are postponed.
+
+1. Release Manager:
+    - Post this checklist in Release Checklist channel
+    - Verify all items in the last posted release checklist are complete
+    - Update Changelog PR based on what’s in/out of the release
+    - Post a reminder to devs in the Desktop App channel of the the code complete date with the ZBB count
```
Comment (deanwhillier) [inline docs/release-process.md]:
```suggestion
    - Post a reminder to devs in the Desktop App channel of the code complete date with the ZBB count
```

File: docs/release-process.md [hunk 2]
```diff
@@ -12,70 +18,83 @@ Notes:
...
+3. Marketing:
+    - Tweet announcement that RC1 is ready
+    
+### C. (T-minus 15 working days) Judgment Day
 
-### C. (T-minus 7 working days) Code Complete and Release Candidate Cut
+Day when Release Manager and PMs decide which major features are included in the release, and which are postponed.
+
+1. Release Manager:
+    - Post this checklist in Release Checklist channel
```
[the remark this answers is not in the document]
Comment (amyblais) [reply docs/release-process.md]: Thanks, updated.

File: docs/release-process.md [hunk 3]
```diff
@@ -12,70 +18,83 @@ Notes:
...
 
-### B. (T-minus 12 working days) Release Date Announcement
+### B. (T-minus 20 working days) Feature Complete
 
-1. PM:
+No pull requests for features should be merged to the current release after this date. In special cases, exceptions can be made by the Release Manager.
+
+1. Release Manager:
     - Post this checklist in Desktop App channel
+    - Follow that any feature PR reviews are prioritized and post a list of outstanding feature PRs in the Desktop App channel
```
[the remark this answers is not in the document]
Comment (amyblais) [reply docs/release-process.md]: Updated to clarify it a little.

**Pull request merged**